doc/bash_completion.txt

```diff
@@ -25,6 +25,28 @@ Environment variables
     loaded. If unset or null, the default compatibility directory to use is
     `/etc/bash_completion.d`.
 
+*BASH_COMPLETION_COMPAT_IGNORE*::
```

Comment:
I realized we haven't quite resolved the naming issue which also relates to these variables in https://github.com/scop/bash-completion/discussions/539. Let's see if we can get that one resolved before merging this.

Reply:
Now #678 has been merged for the naming of the configuration variables, so I have rebased this PR. Now this is ready for review.